src/axes/axis.ts

```diff
@@ -150,9 +168,238 @@ export class Axis<D> extends Component {
                                                     .attr(this._generateTickMarkAttrHash(true));
     tickMarks.exit().remove();
     this._baseline.attr(this._generateBaselineAttrHash());
+    if (this.annotationsEnabled()) {
+      this._drawAnnotations();
+    } else {
+      this._removeAnnotations();
+    }
+    return this;
+  }
+
+  /**
+   * Gets the annotated ticks
+   */
+  public annotatedTicks(): D[];
+  /**
+   * Sets the annotated ticks
+   *
+   * @returns {Axis} The calling Axis.
+   */
+  public annotatedTicks(annotatedTicks: D[]): Axis<D>;
+  public annotatedTicks(annotatedTicks?: D[]): any {
+    if (annotatedTicks == null) {
+      return this._annotatedTicks;
+    }
+    this._annotatedTicks = annotatedTicks;
+    this.render();
+    return this;
+  }
+
+  /**
+   * Gets the formatter for the annotations
+   */
+  public annotationFormatter(): Formatter;
+  /**
+   * Sets the formatter for the annotations
+   *
+   * @returns {Axis} The calling Axis.
+   */
+  public annotationFormatter(annotationFormatter: Formatter): Axis<D>;
+  public annotationFormatter(annotationFormatter?: Formatter): any {
+    if (annotationFormatter == null) {
+      return this._annotationFormatter;
+    }
+    this._annotationFormatter = annotationFormatter;
+    this.render();
+    return this;
+  }
+
+  /**
+   * Gets if annotations are enabled
+   */
+  public annotationsEnabled(): boolean;
+  /**
+   * Sets if annotations are enabled
```

Comment:
I will likely need to update the `margin()` documentation.  Annotations in and of themselves do not request space